src/xarray_regrid/methods/conservative.py

```diff
@@ -107,50 +105,53 @@ def conservative_regrid_dataset(
     """Dataset implementation of the conservative regridding method."""
     data_vars = dict(data.data_vars)
     data_coords = dict(data.coords)
-    valid_fracs = {v: xr.DataArray(name=EMPTY_DA_NAME) for v in data_vars}
     data_attrs = {v: data_vars[v].attrs for v in data_vars}
     coord_attrs = {c: data_coords[c].attrs for c in data_coords}
     ds_attrs = data.attrs
 
+    # Create weights array and coverage mask for each regridding dim
+    weights = {}
+    covered = {}
     for coord in coords:
-        covered_grid = (coords[coord] <= data[coord].max()) & (
+        covered[coord] = (coords[coord] <= data[coord].max()) & (
             coords[coord] >= data[coord].min()
         )
 
         target_coords = coords[coord].to_numpy()
         source_coords = data[coord].to_numpy()
         nd_weights = get_weights(source_coords, target_coords)
 
-        # Modify weights to correct for latitude distortion
-        weights = utils.create_dot_dataarray(
+        da_weights = utils.create_dot_dataarray(
             nd_weights, str(coord), target_coords, source_coords
         )
+        # Modify weights to correct for latitude distortion
         if coord == latitude_coord:
-            weights = apply_spherical_correction(weights, latitude_coord)
-
-        for array in data_vars.keys():
-            if coord in data_vars[array].dims:
-                if sparse is not None:
-                    var_weights = sparsify_weights(weights, data_vars[array])
-                else:
-                    var_weights = weights
-
-                data_vars[array], valid_fracs[array] = apply_weights(
-                    da=data_vars[array],
-                    weights=var_weights,
-                    coord=coord,
-                    valid_frac=valid_fracs[array],
-                    skipna=skipna,
-                )
-                # Mask out any regridded points outside the original domain
-                data_vars[array] = data_vars[array].where(covered_grid)
-
-    if skipna:
-        # Mask out any points that don't meet the nan threshold
-        valid_threshold = get_valid_threshold(nan_threshold)
-        for array, da in data_vars.items():
-            data_vars[array] = da.where(valid_fracs[array] >= valid_threshold)
-
+            da_weights = apply_spherical_correction(da_weights, latitude_coord)
+        weights[coord] = da_weights
+
+    # Apply the weights, using a unique set that matches chunking of each array
```

Comment:
drive-by: in the future, this may be worth reconsidering. xref https://github.com/dask/dask/issues/2225

Reply:
Yeah good point. I realized what I had before will always produce a -1 chunk on the output grid which is going to blow up memory for the long and skinny configuration on a bigger problem.

I added an `output_chunks` kwarg so this is explicitly configurable, matching xesmf. If not provided we use the same size as the input chunks, also matching xesmf. This was a little slower in my tests but better memory footprint. And if you start with spatially chunked data I would expect to get the same format out.